R/exploratoryfactoranalysis.R

```diff
@@ -413,6 +491,70 @@ ExploratoryFactorAnalysis <- function(jaspResults, dataset, options, ...) {
 
 }
 
+# Modification here:
+# Creates a table that outputs the details of the parallel analysis.
+# Particularly useful for reporting the rationale of how many factors were
+# retained - especially within psychometric papers.
+# We are also going to be adding an asterisk aside any factor whose real data eigenvalue is above
+# the 95th percentile simulated data eigenvalue
+.efaPATable <- function(modelContainer, dataset, options, ready) {
+  if (!options[["incl_PAtable"]] || !is.null(modelContainer[["patab"]])) return()
+
+  if (options[["basedOn"]] == "mixed") {
+    polytetracor <- psych::mixedCor(dataset)
+    pa <- try(psych::fa.parallel(polytetracor$rho,
+                                 plot = FALSE,
+                                 fa = options$parallelMethod,
+                                 n.obs = nrow(dataset)))
+  }
+  else {
+    pa <- try(psych::fa.parallel(dataset, plot = FALSE, fa = options$parallelMethod))
+  }
+
+  patab <- createJaspTable(gettext("Parallel Analysis"))
+  patab$dependOn("incl_PAtable")
+  patab$addColumnInfo(name = "col", title = "", type = "string")
+  patab$addColumnInfo(name = "val1", title = gettext("Real Data Eigenvalue"), type = "number", format = "dp:3")
```

Comment:
```suggestion
  patab$addColumnInfo(name = "val1", title = gettext("Real data eigenvalue"), type = "number", format = "dp:3")
```
the column as well as the row names should be lower case